apps/recorder/interface.html

```diff
@@ -1,359 +1,925 @@
+          }
+        });
+
+        // Handle dragging near track line
+        map.on('mousemove', e => {
+          const nearLine = L.GeometryUtil.closest(map, polyline.getLatLngs(), e.latlng)?.distance <= 20;
+          map.dragging[nearLine ? 'disable' : 'enable']();
+          map._container.style.cursor = nearLine ? 'pointer' : '';
+        }).on('mouseout', () => {
+          map.dragging.enable();
+          map._container.style.cursor = '';
+        });
+      }
+
+      // Start/end markers
+      const markerStyle = { radius: 6, color: '#fff', weight: 2, opacity: 1, fillOpacity: 0.8, bubblingMouseEvents: false };
+      const addMarker = (coord, color, text) => {
+        const marker = L.circleMarker(coord, { ...markerStyle, fillColor: color }).addTo(map).bindPopup(text);
+        marker.on('click dblclick mousedown mouseup touchstart touchend', L.DomEvent.stopPropagation);
+        return marker;
+      };
+
+      addMarker(coordinates[0], '#22c55e', startTime ? `Start - ${startTime}` : 'Start');
+      if (coordinates.length > 1) {
+        addMarker(coordinates[coordinates.length - 1], '#ef4444', endTime ? `End - ${endTime}` : 'End');
+      }
+
+      map.fitBounds(polyline.getBounds().pad(0.1));
+    }
+
+    leafletMaps[containerId] = map;
+    return map;
+  } catch (error) {
+    console.error('Error creating map:', error);
+    return null;
+  }
+}
+
+function cleanupMaps(mapId = null) {
+  const removeMaps = m => m?.remove?.();
+  if (mapId && leafletMaps[mapId]) {
+    removeMaps(leafletMaps[mapId]);
+    delete leafletMaps[mapId];
+  }
+  else {
+    Object.values(leafletMaps).forEach(removeMaps);
+    leafletMaps = {};
+  }
+}
+
+function getTrackList() {
+  Util.showModal("Loading Track List...");
+  domTracks.innerHTML = "";
+  cleanupMaps();
+  cleanupCharts();
+
+  Puck.eval(`require("Storage").list(/^recorder\\.log.*\\.csv$/,{sf:1})`,files=>{
+    let trackList = [];
+    // Use promise chain to load tracks sequentially - this prevents overwhelming the Bangle
+    // with too many concurrent requests and ensures stable Bluetooth communication
+    let promise = Promise.resolve();
+
+    files.forEach(filename => {
+      promise = promise.then(()=>new Promise(resolve => {
+        const matches = filename.match(/^recorder\.log(.*)\.csv$/);
+        const trackNo = matches ? matches[1] : '';
+        Util.showModal(`Loading Track ${trackNo}...`);
+        // This function runs on the Bangle to quickly scan for valid GPS data
+        // Many tracks start recording before GPS lock, so we search up to 100 lines
+        // to find the first coordinate for the track preview
+        Puck.eval(`(function(fn) {
+  var f = require("Storage").open(fn,"r");
+  var headers = f.readLine().trim();
+  var data = f.readLine();
+  var lIdx = headers.split(",").indexOf("Latitude");
+  if (lIdx >= 0) {
+    var tries = 100;
+    var l = data;
+    while (l && l.split(",")[lIdx]=="" && tries--)

```

Comment:
Ah, I was wondering why a few routes were no longer shown - this bugfix!

Not one for this PR, but I think once we've fetched the data we could re-decide/scan for lat/lon and then draw routes which have only acquired a GPS fix after 100 attempts